omnigibson/transition_rules.py

```diff
@@ -277,37 +278,42 @@ def transition(self, individual_objects, group_objects):
         t_results = TransitionResults()
 
         # Load object parts.
-        for _, part_idx in enumerate(sliced_obj.metadata["object_parts"]):
-            # List of dicts gets replaced by {'0':dict, '1':dict, ...}.
-            part = sliced_obj.metadata["object_parts"][part_idx]
-            part_category = part["category"]
-            part_model = part["model"]
+        for part_idx, part in sliced_obj.metadata["object_parts"].items():
+            # List of dicts gets replaced by {'0':dict, '1':dict, ...}
+
+            # Get bounding box info
+            part_bb_pos = np.array(part["bb_pos"])
+            part_bb_orn = np.array(part["bb_orn"])
+
+            # Determine the relative scale to apply to the object part from the original object
+            # Note that proper (rotated) scaling can only be applied when the relative orientation of
+            # the object part is a multiple of 90 degrees wrt the parent object, so we assert that here
+            # Check by making sure the real component of rotation is close to 0, 0.707, or 1
+            assert np.any(np.isclose(np.ones(3) * part_bb_orn[-1], np.array([1.0, 0.7071, 0.0]), atol=1e-3)), \
```

Comment:
hmm I am not sure this is the best way to compare two orientations. Can we use a more principled method?

Reply:
do you black magic

Reply:
black magic done